packages/ember-data/lib/system/store.js

```diff
@@ -1045,18 +1045,11 @@ Store = Service.extend({
   */
   all: function(modelName) {
     var typeClass = this.modelFor(modelName);
-    var typeMap = this.typeMapFor(typeClass);
-    var findAllCache = typeMap.findAllCache;
 
-    if (findAllCache) {
-      this.recordArrayManager.updateFilter(findAllCache, typeClass);
-      return findAllCache;
-    }
-
-    var array = this.recordArrayManager.createRecordArray(typeClass);
+    var liveRecordArray = this.recordArrayManager.liveRecordArrayFor(typeClass);
+    this.recordArrayManager.updateFilter(liveRecordArray, typeClass);
```

Comment:
This call *could* move inside the `defaultValue` function for the `liveRecordArrays` map, but then [this spec](https://github.com/emberjs/data/blob/58c2cf65eb492165c56ba561803357b99020305a/packages/ember-data/tests/integration/all-test.js#L36-L46) fails...